### PR TITLE
Cross compile fix

### LINKED ISSUE #190: Windows build fails
```
[release : step-release] vendor/github.com/kr/pty/shim.go:26:16: undefined: pty.Winsize
[release : step-release] vendor/github.com/kr/pty/shim.go:32:63: undefined: pty.Getsize
[release : step-release] vendor/github.com/kr/pty/shim.go:38:9: undefined: pty.GetsizeFull
[release : step-release] vendor/github.com/kr/pty/shim.go:47:52: undefined: pty.InheritSize
[release : step-release] vendor/github.com/kr/pty/shim.go:57:54: undefined: pty.Setsize
[release : step-release] vendor/github.com/kr/pty/shim.go:64:60: undefined: pty.Start
[release : step-release] vendor/github.com/kr/pty/shim.go:75:9: undefined: pty.StartWithSize
```

We need to fix that ! And we need to run those builds on the CI !

/assign
/kind bug
/priority critical-urgent

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

- Add a `cross` Makefile target
- Configure the CI to use it in `build_test`
- Use `github.com/creack/pty` instead of `github.com/kr/pty`.

github.com/kr/pty doesn't support multiple platform and is now
archived. github.com/creack/pty is the new home for this repository.

As it's a transitive dependency, we don't have control over the
dependency *but* we can use the `replace` feature to force go to use
creack/pty instead of kr/pty for this repository.

Fixes #190 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -60,3 +60,5 @@ require (
 	k8s.io/kubernetes v1.13.3 // indirect
 	sigs.k8s.io/yaml v1.1.0 // indirect
 )
+
+replace github.com/kr/pty => github.com/creack/pty v1.1.7
```
Comment (vdemeester) [inline go.mod]: This is the key part of this PR (with the `Makefile` and the `presumbit-tests.sh` update)

File: Makefile [hunk 2]
```diff
@@ -5,6 +5,30 @@ FORCE:
 ./vendor: go.mod go.sum
 	@go mod vendor
 
+.PHONY: cross
+cross: amd64 386 arm arm64 ## build cross platform binaries
+
+.PHONY: amd64
+amd64:
+	GOOS=linux GOARCH=amd64 go build -o bin/tkn-linux-amd64 ./cmd/tkn
```
Comment (sthaha) [inline Makefile]: shouldn't we set the `-ldflags ` for `version`?

Comment (vdemeester) [reply Makefile]: @sthaha not really, those binaries are not published that way, this is there just to be able to cross-build

**Pull request merged**